### PR TITLE
tests for `src/lib/resolvers/Subscription/messageSentToGroupChat.ts`

### LINKED ISSUE #889: Test: src/lib/resolvers/Subscription/messageSentToGroupChat.ts
- Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue https://github.com/PalisadoesFoundation/talawa/issues/359

The Talawa-API code base needs to be 100% reliable. This means we need to have 100% test code coverage.

- Tests need to be written for file `src/lib/resolvers/Subscription/messageSentToGroupChat.ts`
- We will need the API to be refactored for all methods, classes and/or functions found in this file for testing to be correctly executed.
- When complete, all all methods, classes and/or functions in the refactored file will need to be tested. These tests must be placed in a
single file with the name `__tests__/resolvers/Subscription/messageSentToGroupChat.spec.ts`. You may need to create the appropriate directory structure to do this.

### IMPORTANT:
Please refer to the parent issue on how to implement these tests correctly:

- https://github.com/PalisadoesFoundation/talawa-api/issues/490

### PR Acceptance Criteria

- When complete this file must show **100%** coverage when merged into the code base. This will be clearly visible when you submit your PR.
- [The current code coverage for the file can be found by visting this page](https://app.codecov.io/gh/PalisadoesFoundation/talawa-api?displayType=list). Login using your GitHub credentials.
- Create your own `codecov.io` to help with testing.
- The PR you create will show a report for the code coverage for the file you have added. You can also use that as a guide.

### PR DESCRIPTION
**Pull request by KrutikaBhatt**
**What kind of change does this PR introduce?**

**Issue Number:** #889 

Fixes #889 

**Description**:
Written tests for messageSentToGroupChat subscription by creating a mock test GroupChat Message.

**Did you add tests for your changes? : Yes**

**Snapshots/Videos:**
The coverage obtained is 100%
image

**Other information**
The subscription uses `withFilter` which has a filter function parameter that returns true if the user in the GroupChat is currentUser. The code block is the same just converted into a separate function for testing and used in the subscription 

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)? Yes**

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@KrutikaBhatt Please refer to the PR_GUIDELINES.md document in the root folder of the repo. It tells you how to automatically close your issue when the PR is merged.

**Comment by KrutikaBhatt** [comment]
> @KrutikaBhatt Please refer to the PR_GUIDELINES.md document in the root folder of the repo. It tells you how to automatically close your issue when the PR is merged.

Yes, sure! Will follow

**Review by noman2002 — COMMENTED** [review]
Why package-lock.json is removed ? @KrutikaBhatt

**Comment by KrutikaBhatt** [comment]
No, it shouldn't be removed; it might be deleted by mistake

File: package-lock.json [hunk 1]
```diff
@@ -24212,4 +24212,4 @@
       }
     }
   }
-}
+}
```
Comment (noman2002) [inline package-lock.json]: Fix this. There is and extra parenthesis.

**Review by noman2002 — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (KrutikaBhatt) [reply package-lock.json]: Yes, I restored the original package-lock.json

**Comment by KrutikaBhatt** [comment]
##### The only branch not covered - 

A possible solution to it is 

```
import { PubSub } from 'graphql-subscriptions';
const pubsub = new PubSub();

.....
subscribe: withFilter(
      (_parent, _args, context) =>
        pubsub.asyncIterator([MESSAGE_SENT_TO_GROUP_CHAT]),
```

##### rather than using
```context.pubsub.asyncIterator([MESSAGE_SENT_TO_GROUP_CHAT])```



Any views on this ?

**Comment by KrutikaBhatt** [comment]
@noman2002, I have covered the remaining branch also. The code coverage is shown below - 

image

**Review by noman2002 — CHANGES_REQUESTED** [review]
package-lock.json is still not restored. please restore.

**Comment by KrutikaBhatt** [comment]
> package-lock.json is still not restored. please restore.

I just copied the package-lock.json from main file. I will just create a new pull request with the same tests. to avoid any problems with package-lock.json

**Comment by KrutikaBhatt** [comment]
I am closing this PR. The same tests are covered in new PR - #1113

**Pull request closed without merging**